gnpy/core/request.py

```diff
@@ -115,149 +115,77 @@ def __init__(self,path_request,computed_path):
-                           'accumulative-value': 'None'
-                           },
-                           {
-                           'metric-type': 'OSNR@0.1nm',
-                           'accumulative-value': 'None'
-                           },
-                           {
-                           'metric-type': 'reference_power',
-                           'accumulative-value': self.path_request.power
-                           },
-                           {
-                           'metric-type': 'path_bandwidth',
-                           'accumulative-value': self.path_request.path_bandwidth
-                           }
-                        ],
-                        'path-srlgs': {
-                            'usage': 'not used yet',
-                            'values': 'not used yet'
-                        },
-                        'path-route-objects': [
-                            {
-                            'path-route-object': {
-                                'index': 0,
-                                'unnumbered-hop': {
-                                    'node-id': self.path_request.source,
-                                    'link-tp-id': self.path_request.source,
-                                    'hop-type':  self.hop_type,
-                                    'direction': 'not used'
-                                },
-                                'label-hop': {
-                                    'te-label': {
-                                        'generic': 'not used yet',
-                                        'direction': 'not used yet'
-                                        }
-                                    }
-                                }
-                            },
-                            {
-                            'path-route-object': {
-                                'index': 1,
-                                'unnumbered-hop': {
-                                    'node-id': self.path_request.destination,
-                                    'link-tp-id': self.path_request.destination,
-                                    'hop-type':  self.hop_type,
-                                    'direction': 'not used'
-                                },
-                                'label-hop': {
-                                    'te-label': {
-                                        'generic': 'not used yet',
-                                        'direction': 'not used yet'
-                                        }
-                                    }
-                                }
+                   'response-id': self.path_id,
+                   'no-path': "Response without path information, due to failure performing the path computation"
+                   }
+        else:
+            index = 0
+            pro_list = []
+            for nel in self.computed_path:
```

Comment:
I do not think that changes such as the one done in 8f3a468527155f971537676265c1694369aac6b1 are good. There are contexts where `n` is a perfectly good name. I think that this is not one of them, and because this iteration is about an element within a path, why don't you call it just `element`?

Reply:
the changes in [https://github.com/Telecominfraproject/oopt-gnpy/commit/8f3a468527155f971537676265c1694369aac6b1](url) has been made because of codacy report: to conform to '[a-z_][a-z0-9_]{2,30}$' pattern. I believe this is a good rule to follow. I wanted to avoid naming all my lists variables with the same name.
I will change to element.

Reply:
[https://github.com/Telecominfraproject/oopt-gnpy/pull/277#discussion_r331810545](url) you say

>I do not understand. These data are present in the CSV file, and you said that you're using the panda library so that you can have per-column access to the data sets -- right? What's the point in copying that info in here? I'm not saying that I am right, I'm asking what I am missing.

This is test for the jsontocsv function.

The expected file contains the result that is expected in a csv created based on the json answer `jsontocsv(json_data, equipment, fcsv)`.
the test checks that 1) the jsontocsv correctly transform the json into csv in terms of parameters names (header) 2) in terms of values (same results) 3) for all requests, whatever the order of requests, or the order of columns.